### PR TITLE
Speedup by removing non pattern replaceAll with constant arg

### PR DESCRIPTION
**Pull request by tbw777**
replaceAll("ABC", "") is non Pattern method and therefore must be replaced to simple fast replace()
A proofs of changes: https://gist.github.com/tbw777/8a6ef60af21487c5faec67037099fd0b

### CODE REVIEW

File: java/org/apache/tomcat/buildutil/translate/Utils.java [hunk 1]
```diff
@@ -123,7 +123,7 @@ static String formatValueCommon(String in) {
 
         result = ESCAPE_LEADING_SPACE.matcher(result).replaceAll("\\\\$1");
 
-        result = result.replaceAll("\t", "\\t");
+        result = result.replace("\t", "t");
```
Comment (markt-asf) [inline java/org/apache/tomcat/buildutil/translate/Utils.java:127]: This is not correct. The literal (unescaped) replacement text should be `\t`

**Review by markt-asf — CHANGES_REQUESTED** [review]
Thanks for this PR - these are good changes to have.

File: java/org/apache/tomcat/buildutil/translate/Utils.java (hunk 1 above)
Comment (tbw777) [reply java/org/apache/tomcat/buildutil/translate/Utils.java:127]: The second argument in replaceAll means escape t, not \t.
You can check proofs tests. equals returning true if replace maked to t, not \t

Comment (markt-asf) [reply java/org/apache/tomcat/buildutil/translate/Utils.java:127]: It is a bug in the original code. It should be replacing all tabs with the literal string `\t`.

Comment (markt-asf) [reply java/org/apache/tomcat/buildutil/translate/Utils.java:127]: For the record, none of the current i18n strings use tabs. I did find some messages using them in early 7.0.x versions, but those messages were never referenced in the Java code.

**Comment by tbw777** [comment]
@markt-asf 
Branch was updated and https://gist.github.com/tbw777/8a6ef60af21487c5faec67037099fd0b also
Check please.

File: java/org/apache/tomcat/buildutil/translate/Utils.java (hunk 1 above)
Comment (markt-asf) [reply java/org/apache/tomcat/buildutil/translate/Utils.java:127]: Tx.

**Review by markt-asf — APPROVED** [review]

**Comment by ChristopherSchultz** [comment]
There isn't much of a difference between `String.replace` and `String.replaceAll`. They both use regular expressions under the hood. Java's `Pattern` class has a special case for "this search-string doesn't include any metacharacters" (it's a flag called `LITERAL`) and has optimized code-paths for those cases.

IMO `replaceAll` is more clear than `replace` because it's absolutely clear that all instances will be replaced in the string.

If performance is the primary factor, here, we should use an implementation of string-replacement that does not involve creating `Pattern` and `Matcher` objects and re-encoding the replacement String _every single time_.

**Comment by ChristopherSchultz** [comment]
I think this "performance improvement" is not an improvement and doesn't affect performance in any meaningful way.

**Comment by ChristopherSchultz** [comment]
Apologies: my comments are rubbish after Java 9. Please forgive the noise.

**Comment by happyhua** [comment]
A replaceAll -> replace change in for example `java/org/apache/tomcat/buildutil/translate/Utils.java` can produce different output.


```
        String source = "1\t2\t3";

        System.out.println("source: " + source);
        System.out.println("replaceAll: " + source.replaceAll("\t", "\\t"));
        System.out.println("replace: " + source.replace("\t", "\\t"));

```

```
source: 1	2	3
replaceAll: 1t2t3
replace: 1\t2\t3
```

**Pull request merged**